repos/system_upgrade/el7toel8/actors/checksendmail/libraries/library.py

```diff
@@ -0,0 +1,27 @@
+import os
+import re
+
+SendmailConfDir = '/etc/mail'
+SendmailConfFiles = ['sendmail.cf', 'sendmail.mc', 'submit.cf', 'submit.mc']
+# false positives blacklist
+rfp = re.compile(r'(^\s*RIPv6:::1\b)|(@\s+\[IPv6:::1\]\s+>)')
+
+conf_dir = os.path.normpath(SendmailConfDir)
+conf_files = [os.path.join(conf_dir, f) for f in SendmailConfFiles]
+
+def check_false_positives(filename, line):
+    return filename in ['sendmail.cf', 'submit.cf'] and rfp.search(line) is not None
+
+def check_files_for_compressed_ipv6():
+    migrate_files = []
+    files = [os.path.join(conf_dir, re.sub(r'\.db$', '', f)) for f in os.listdir(conf_dir) if f.endswith('.db')] + conf_files
```

Comment:
To prevent the exception from happening and to make the code more robust here:
```suggestion
    files = []
    if os.path.isdir(conf_dir):
        files = [os.path.join(conf_dir, re.sub(r'\.db$', '', f)) for f in os.listdir(conf_dir) if f.endswith('.db')] + conf_files
```